frontend/src/common/MarkdownDisplay.tsx

```diff
@@ -17,31 +18,34 @@ export default function MarkdownDisplay({ children }: MarkdownDisplayProps) {
     () => ({
       overrides: {
         h1: {
-          component: Typography,
-          props: {
-            gutterBottom: true,
-            variant: 'h4',
-          },
+          component: (props: TypographyProps) => (
+            <Typography gutterBottom variant='h4' component='h1' sx={{ wordWrap: 'break-word' }} {...props} />
+          ),
         },
         h2: {
-          component: Typography,
-          props: { gutterBottom: true, variant: 'h6' },
+          component: (props: TypographyProps) => (
+            <Typography gutterBottom variant='h6' component='h2' sx={{ wordWrap: 'break-word' }} {...props} />
+          ),
         },
         h3: {
-          component: Typography,
-          props: { gutterBottom: true, variant: 'subtitle1' },
+          component: (props: TypographyProps) => (
+            <Typography gutterBottom variant='subtitle1' component='h2' sx={{ wordWrap: 'break-word' }} {...props} />
+          ),
         },
         h4: {
-          component: Typography,
-          props: {
-            gutterBottom: true,
-            variant: 'caption',
-            paragraph: true,
-          },
+          component: (props: TypographyProps) => (
+            <Typography
+              gutterBottom
+              paragraph
+              variant='caption'
+              component='h2'
+              sx={{ wordWrap: 'break-word' }}
+              {...props}
+            />
+          ),
         },
         p: {
-          component: Typography,
-          props: { paragraph: true },
+          component: (props: TypographyProps) => <Typography sx={{ wordWrap: 'break-word' }} {...props} />,
         },
```

Comment:
To fix the wrapping for a single paragraph you'll need to add the following to override the styling of `span`s:
```
        span: {
          component: (props: TypographyProps) => (
            <Typography component='span' sx={{ wordWrap: 'break-word' }} {...props} />
          ),
        },
```